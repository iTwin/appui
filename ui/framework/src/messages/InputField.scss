.nz-popup-message-inputField {
  margin: 5px 0;
  transition: all 3s ease;
}

.message-inputField-content {
  background-color: #ffffff;
}
