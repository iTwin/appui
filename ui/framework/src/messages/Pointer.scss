/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/

@import "@bentley/ui-core/lib/colors";
@import "@bentley/bwc/lib/mixins";

.nz-popup-message-pointer {
  background-color: $background-1;
  padding: 5px 15px;
  border-radius: 15px;
  box-shadow: 1px 1px 5px 1px $background-5;
  color: $text-dark;
}

.tooltip {
  text-decoration:none;
  position:relative;
}

.tooltip span {
  display:none;
}

.tooltip:hover span {
  display:block;
  position:fixed;
  overflow:visible;
}

.popup-message-brief {
  @include bwc-text(body);
}

.popup-message-detailed {
  @include bwc-text(small);
}