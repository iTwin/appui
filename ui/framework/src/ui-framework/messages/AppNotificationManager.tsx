/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Notification */

import {
  ActivityMessageDetails,
  ActivityMessageEndReason,
  MessageBoxIconType,
  MessageBoxType,
  MessageBoxValue,
  NotificationManager,
  NotifyMessageDetails,
  ToolTipOptions,
  OutputMessageType,
} from "@bentley/imodeljs-frontend";

import { XAndY } from "@bentley/geometry-core";

import { MessageManager } from "./MessageManager";
import { UiFramework } from "../UiFramework";
import { ElementTooltip } from "../feedback/ElementTooltip";
import PointerMessage from "./Pointer";

/**
 * The AppNotificationManager class is a subclass of NotificationManager. This implementation uses
 * the iModel.js UI library to display alerts, messages, prompts and tooltips.
 *
 * The NotificationManager controls the interaction with the user for prompts, error messages, and alert dialogs.
 * Implementations of the NotificationManager may present the information in different ways. For example, in
 * non-interactive sessions, these messages may be saved to a log file or simply discarded.
 */
export class AppNotificationManager extends NotificationManager {

  /** Output a prompt, given an i18n key. */
  public outputPromptByKey(key: string): void {
    this.outputPrompt(UiFramework.i18n.translate(key));
  }

  /** Output a prompt to the user. A 'prompt' indicates an action the user should take to proceed. */
  public outputPrompt(prompt: string): void {
    MessageManager.outputPrompt(prompt);
  }

  /** Output a message and/or alert to the user. */
  public outputMessage(message: NotifyMessageDetails): void {
    if (message.msgType === OutputMessageType.Pointer)
      this._showPointerMessage(message);
    MessageManager.addMessage(message);
  }

  /** Output a MessageBox and wait for response from the user.
   * @param mbType       The MessageBox type.
   * @param message      The message to display.
   * @param icon         The MessageBox icon type.
   * @return the response from the user.
   */
  public async openMessageBox(mbType: MessageBoxType, message: string, icon: MessageBoxIconType): Promise<MessageBoxValue> {
    return MessageManager.openMessageBox(mbType, message, icon);
  }

  /**
   * Set up for activity messages.
   * @param details  The activity message details.
   * @return true if the message was displayed, false if an invalid priority is specified.
   */
  public setupActivityMessage(details: ActivityMessageDetails): boolean {
    return MessageManager.setupActivityMessageDetails(details);
  }

  /**
   * Output an activity message to the user.
   * @param messageText      The message text.
   * @param percentComplete  The percentage of completion.
   * @return true if the message was displayed, false if the message could not be displayed.
   */
  public outputActivityMessage(messageText: string, percentComplete: number): boolean {
    return MessageManager.setupActivityMessageValues(messageText, percentComplete);
  }

  /**
   * End an activity message.
   * @param reason       Reason for the end of the Activity Message.
   * @return true if the message was ended successfully, false if the activityMessage could not be ended.
   */
  public endActivityMessage(reason: ActivityMessageEndReason): boolean {
    switch (reason) {
      case (ActivityMessageEndReason.Completed):
        return MessageManager.endActivityMessage(true);
      case (ActivityMessageEndReason.Cancelled):
        return MessageManager.endActivityMessage(false);
    }
    return false;
  }

  /** Hides the Pointer message. */
  public closePointerMessage(): void { this._hidePointerMessage(); }

  /** Return true if _showTooltip has an implementation and will display a tooltip. */
  public get isToolTipSupported(): boolean { return true; }

  /** Return true if the tooltip is currently open. */
  public get isToolTipOpen(): boolean { return ElementTooltip.isTooltipVisible; }

  /** Clear the ToolTip if it is currently open. If not open, does nothing. */
  public clearToolTip(): void {
    if (this.isToolTipOpen)
      ElementTooltip.hideTooltip();
  }

  /** Show a ToolTip window.
   * @param el       The HTMLElement that that anchors the toolTip.
   * @param message  The message to display inside the ToolTip
   * @param pt       An optional location, relative to the origin of el, for the ToolTip. If undefined, center of el.
   * @param options  Options that supply additional information about how the ToolTip should function.
   */
  protected _showToolTip(el: HTMLElement, message: HTMLElement | string, pt?: XAndY, options?: ToolTipOptions): void {
    ElementTooltip.showTooltip(el, message, pt, options);
  }

  /**
   * Show a Pointer message.
   * @param message  Text to display in message.
   */
  protected _showPointerMessage(message: NotifyMessageDetails): void {
    PointerMessage.showMessage(message);
  }

  /** Hide a Pointer message. */
  protected _hidePointerMessage(): void {
    PointerMessage.hideMessage();
  }
}
