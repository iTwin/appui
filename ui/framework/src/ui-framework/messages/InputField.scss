/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/

.nz-popup-message-inputField {
  margin: 5px 0;
  transition: all 3s ease;
}

.message-inputField-content {
  background-color: #ffffff;
}
