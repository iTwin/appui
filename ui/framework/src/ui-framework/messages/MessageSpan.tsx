/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Notification
 */

import * as React from "react";
import { ClassNameProps } from "@bentley/ui-core";
import { isHTMLElement, isReactMessage, NotifyMessageType } from "./ReactNotifyMessageDetails";

/** @internal */
export interface MessageSpanProps extends ClassNameProps {
  message: NotifyMessageType;
}

/** @internal */
export function MessageSpan(props: MessageSpanProps) {
  let messageNode = null;

  if (typeof props.message === "string")
    messageNode = <span className={props.className}>{props.message}</span>;
  else if (isHTMLElement(props.message))
    messageNode = <span className={props.className} dangerouslySetInnerHTML={{ __html: props.message.outerHTML }} />;
  else if (isReactMessage(props.message))
    messageNode = <span className={props.className}>{props.message.reactNode}</span>;

  return messageNode;
}

/** @internal */
export function MessageDiv(props: MessageSpanProps) {
  let messageNode = null;

  if (typeof props.message === "string")
    messageNode = <div className={props.className}>{props.message}</div>;
  else if (isHTMLElement(props.message))
    messageNode = <div className={props.className} dangerouslySetInnerHTML={{ __html: props.message.outerHTML }} />;
  else if (isReactMessage(props.message))
    messageNode = <div className={props.className}>{props.message.reactNode}</div>;

  return messageNode;
}
