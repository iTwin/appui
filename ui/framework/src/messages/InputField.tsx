/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Notification */

import * as React from "react";
import * as ReactDOM from "react-dom";
import * as classnames from "classnames";
import "./InputField.scss";
import { Div, withOnOutsideClick } from "@bentley/ui-core";
import CommonProps from "@bentley/ui-ninezone/lib/utilities/Props";
import MessageButton from "@bentley/ui-ninezone/lib/footer/message/content/Button";
import Status from "@bentley/ui-ninezone/lib/footer/message/content/status/Status";
import { StatusMessage } from "@bentley/ui-ninezone/lib/footer/message/content/status/Message";
import StatusMessageLayout from "@bentley/ui-ninezone/lib/footer/message/content/status/Layout";

// tslint:disable-next-line:variable-name
const DivWithOnOutsideClick = withOnOutsideClick(Div);

/** Properties of [[InputFieldMessage]] component. */
export interface InputFieldMessageProps extends CommonProps {
  /** Parent of message. */
  target: Element;
  /** Message content. */
  children: React.ReactNode;
  /** Function that will close the message */
  onClose: () => void;
}

/** InputField message is a popup error message that appears under invalid user input. */
export class InputFieldMessage extends React.Component<InputFieldMessageProps> {
  public render(): React.ReactNode {
    return ReactDOM.createPortal(this._getErrorMessage(), this.props.target);
  }

  /**
   * Provides a message to display inside of the portal.
   */
  private _getErrorMessage(): React.ReactNode {
    const className = classnames(
      "nz-popup-message-inputField",
      this.props.className);

    return (
      <DivWithOnOutsideClick
        className={className}
        style={this.props.style}
        // TODO: dismiss onOutsideClick without immediately dismissing message
        children={
          < StatusMessage
            className="nz-popup-message-inputField"
            status={Status.Error}
            icon={
              < i className="icon icon-status-error-hollow" />
            }
          >
            <StatusMessageLayout
              className="message-inputField-content"
              label={this.props.children}
              buttons={
                <MessageButton onClick={this.props.onClose}>
                  <i className="icon icon-close" />
                </MessageButton>
              }
            />
          </StatusMessage>
        }
      />
    );
  }
}

export default InputFieldMessage;
