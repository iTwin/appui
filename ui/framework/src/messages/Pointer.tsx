/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Notification */

import * as React from "react";
import * as classnames from "classnames";
import { UiEvent } from "@bentley/ui-core";
import { XAndY } from "@bentley/geometry-core";
import CommonProps from "@bentley/ui-ninezone/lib/utilities/Props";

import ToolSettingsTooltip from "@bentley/ui-ninezone/lib/widget/tool-settings/Tooltip";

import "./Pointer.scss";
import Tooltip, { offsetAndContainInContainer } from "@bentley/ui-ninezone/lib/popup/tooltip/Tooltip";
import { RelativePosition, NotifyMessageDetails } from "@bentley/imodeljs-frontend";

/** Properties of [[PointerMessage]] component. */
export interface PointerMessageProps extends CommonProps {
  /** Text to display */
  message?: string;
}

/** [[PointerMessage]] state.
 */
export interface PointerMessageState {
  isVisible: boolean;
  message: string;
  detailedMessage?: string;
  relativePosition?: RelativePosition;
  viewport?: HTMLElement;
  pt?: XAndY;
}

/** [[PointerMessageChangedEvent]] arguments.
 */
export interface PointerMessageChangedEventArgs {
  isVisible: boolean;
  message: string;
  detailedMessage?: string;
  relativePosition?: RelativePosition;
  viewport?: HTMLElement;
  pt?: XAndY;
}

/** Pointer Message Changed Event emitted by the [[PointerMessage]] component
 */
export class PointerMessageChangedEvent extends UiEvent<PointerMessageChangedEventArgs> { }

/** Pointer message pops up near pointer when attempting an invalid interaction. */
export class PointerMessage extends React.Component<PointerMessageProps> {
  private static _pointerMessageChangedEvent: PointerMessageChangedEvent = new PointerMessageChangedEvent();

  public static get onPointerMessageChangedEvent(): PointerMessageChangedEvent { return PointerMessage._pointerMessageChangedEvent; }

  public static showMessage(message: NotifyMessageDetails): void {
    PointerMessage.onPointerMessageChangedEvent.emit({
      isVisible: true,
      message: message.briefMessage,
      detailedMessage: message.detailedMessage,
      relativePosition: message.relativePosition,
      viewport: message.viewport,
      pt: message.displayPoint,
    });
  }

  public static hideMessage(): void {
    PointerMessage.onPointerMessageChangedEvent.emit({
      isVisible: false,
      message: "",
    });
  }

  public readonly state: Readonly<PointerMessageState> = {
    message: "",
    isVisible: false,
  };

  public render(): React.ReactNode {
    if (!this.state.isVisible)
      return null;

    const className = classnames(
      "nz-popup-message-pointer",
      this.props.className);

    return (
      <ToolSettingsTooltip
        className={className}
        style={this.props.style}
        position={this.state.pt}
        containIn={this._handleContainIn}
        adjustPosition={this._adjustTooltipPosition()}
      >
        {
          this.state.message &&
          <div
            className="popup-message-brief"
            dangerouslySetInnerHTML={{ __html: this.state.message }}
          />
        }
        {
          this.state.detailedMessage &&
          <div
            className="popup-message-detailed"
            dangerouslySetInnerHTML={{ __html: this.state.detailedMessage }}
          />
        }
      </ToolSettingsTooltip>
    );
  }

  public componentDidMount(): void {
    PointerMessage.onPointerMessageChangedEvent.addListener(this._handlePointerMessageChangedEvent);
  }

  public componentWillUnmount(): void {
    PointerMessage.onPointerMessageChangedEvent.removeListener(this._handlePointerMessageChangedEvent);
  }

  private _handleContainIn = (message: HTMLElement) => {
    if (this.state.viewport)
      return this.state.viewport;
    return Tooltip.defaultProps.containIn(message);
  }

  private _handlePointerMessageChangedEvent = (args: PointerMessageChangedEventArgs) => {
    this.setState(() => ({
      isVisible: args.isVisible,
      message: args.message,
      detailedMessage: args.detailedMessage,
      relativePosition: args.relativePosition,
      viewport: args.viewport,
      pt: args.pt,
    }));
  }

  private _adjustTooltipPosition(): any {
    let tooltipAdjustment = offsetAndContainInContainer();

    const adjustmentOffset = 50;
    switch (this.state.relativePosition) {
      case RelativePosition.Top:
        tooltipAdjustment = offsetAndContainInContainer({ x: 0, y: -adjustmentOffset });
        break;
      case RelativePosition.TopRight:
        tooltipAdjustment = offsetAndContainInContainer({ x: adjustmentOffset, y: -adjustmentOffset });
        break;
      case RelativePosition.Right:
        tooltipAdjustment = offsetAndContainInContainer({ x: adjustmentOffset, y: 0 });
        break;
      case RelativePosition.BottomRight:
        tooltipAdjustment = offsetAndContainInContainer({ x: adjustmentOffset, y: adjustmentOffset });
        break;
      case RelativePosition.Bottom:
        tooltipAdjustment = offsetAndContainInContainer({ x: 0, y: adjustmentOffset });
        break;
      case RelativePosition.BottomLeft:
        tooltipAdjustment = offsetAndContainInContainer({ x: -adjustmentOffset, y: adjustmentOffset });
        break;
      case RelativePosition.Left:
        tooltipAdjustment = offsetAndContainInContainer({ x: -adjustmentOffset, y: 0 });
        break;
      case RelativePosition.TopLeft:
        tooltipAdjustment = offsetAndContainInContainer({ x: -adjustmentOffset, y: -adjustmentOffset });
        break;
    }
    return tooltipAdjustment;
  }
}

export default PointerMessage;
