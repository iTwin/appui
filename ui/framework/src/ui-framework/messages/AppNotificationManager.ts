/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Notification */

import {
  ActivityMessageDetails,
  ActivityMessageEndReason,
  MessageBoxIconType,
  MessageBoxType,
  MessageBoxValue,
  NotificationManager,
  NotifyMessageDetails,
  ToolTipOptions,
  OutputMessageType,
  ToolAssistanceInstructions,
  ToolAssistance,
  RelativePosition,
} from "@bentley/imodeljs-frontend";

import { XAndY } from "@bentley/geometry-core";

import { MessageManager } from "./MessageManager";
import { UiFramework } from "../UiFramework";
import { ElementTooltip } from "../feedback/ElementTooltip";
import { PointerMessage } from "./Pointer";

/**
 * The AppNotificationManager class is a subclass of NotificationManager in @bentley/imodeljs-frontend.
 * This implementation uses the iModel.js UI library to display alerts, messages, prompts and tooltips.
 * @public
 */
export class AppNotificationManager extends NotificationManager {

  /** Output a prompt, given an i18n key.
   */
  public outputPromptByKey(key: string): void {
    this.outputPrompt(UiFramework.i18n.translate(key));
  }

  /** Output a prompt to the user. A 'prompt' indicates an action the user should take to proceed.
   */
  public outputPrompt(prompt: string): void {
    MessageManager.outputPrompt(prompt);

    const mainInstruction = ToolAssistance.createInstruction("", prompt);
    const instructions = ToolAssistance.createInstructions(mainInstruction);
    MessageManager.setToolAssistance(instructions);
  }

  /** Output a message and/or alert to the user. */
  public outputMessage(message: NotifyMessageDetails): void {
    if (message.msgType === OutputMessageType.Pointer) {
      PointerMessage.showMessage(message);
    } else if (message.msgType === OutputMessageType.InputField && message.inputField) {
      MessageManager.displayInputFieldMessage(message.inputField, message.briefMessage, message.detailedMessage, message.priority);
    }
    MessageManager.addMessage(message);
  }

  /** Output a MessageBox and wait for response from the user.
   * @param mbType       The MessageBox type.
   * @param message      The message to display.
   * @param icon         The MessageBox icon type.
   * @return the response from the user.
   */
  public async openMessageBox(mbType: MessageBoxType, message: HTMLElement | string, icon: MessageBoxIconType): Promise<MessageBoxValue> {
    return MessageManager.openMessageBox(mbType, message, icon);
  }

  /**
   * Set up for activity messages.
   * @param details  The activity message details.
   * @return true if the message was displayed, false if an invalid priority is specified.
   */
  public setupActivityMessage(details: ActivityMessageDetails): boolean {
    return MessageManager.setupActivityMessageDetails(details);
  }

  /**
   * Output an activity message to the user.
   * @param messageText      The message text.
   * @param percentComplete  The percentage of completion.
   * @return true if the message was displayed, false if the message could not be displayed.
   */
  public outputActivityMessage(messageText: HTMLElement | string, percentComplete: number): boolean {
    return MessageManager.setupActivityMessageValues(messageText, percentComplete);
  }

  /**
   * End an activity message.
   * @param reason       Reason for the end of the Activity Message.
   * @return true if the message was ended successfully, false if the activityMessage could not be ended.
   */
  public endActivityMessage(reason: ActivityMessageEndReason): boolean {
    let result = false;

    switch (reason) {
      case ActivityMessageEndReason.Completed:
        result = MessageManager.endActivityMessage(true);
        break;
      case ActivityMessageEndReason.Cancelled:
        result = MessageManager.endActivityMessage(false);
        break;
    }

    return result;
  }
  /** Update message position created with [[OutputMessageType.Pointer]].
   * @param displayPoint        Point at which to display the Pointer type message.
   * @param relativePosition    Position relative to displayPoint at which to display the Pointer type message.
   */
  public updatePointerMessage(displayPoint: XAndY, relativePosition: RelativePosition): void {
    PointerMessage.updateMessage(displayPoint, relativePosition);
  }

  /** Hides the Pointer message. */
  public closePointerMessage(): void {
    PointerMessage.hideMessage();
  }

  /** Return true if _showTooltip has an implementation and will display a tooltip. */
  public get isToolTipSupported(): boolean { return true; }

  /** Return true if the tooltip is currently open. */
  public get isToolTipOpen(): boolean { return ElementTooltip.isTooltipVisible; }

  /** Clear the ToolTip if it is currently open. If not open, does nothing. */
  public clearToolTip(): void {
    // istanbul ignore else
    if (this.isToolTipOpen)
      ElementTooltip.hideTooltip();
  }

  /** Show a ToolTip window.
   * @param el       The HTMLElement that anchors the toolTip.
   * @param message  The message to display inside the ToolTip
   * @param pt       An optional location, relative to the origin of el, for the ToolTip. If undefined, center of el.
   * @param options  Options that supply additional information about how the ToolTip should function.
   */
  protected _showToolTip(el: HTMLElement, message: HTMLElement | string, pt?: XAndY, options?: ToolTipOptions): void {
    ElementTooltip.showTooltip(el, message, pt, options);
  }

  /** Hide a InputField message. */
  public closeInputFieldMessage(): void {
    MessageManager.hideInputFieldMessage();
  }

  /** Setup tool assistance instructions for a tool. The instructions include the main instruction, which includes the current prompt.
   * @param instructions The tool assistance instructions.
   * @alpha
   */
  public setToolAssistance(instructions: ToolAssistanceInstructions | undefined) {
    MessageManager.outputPrompt(instructions ? instructions.mainInstruction.text : "");
    MessageManager.setToolAssistance(instructions);
  }

}
