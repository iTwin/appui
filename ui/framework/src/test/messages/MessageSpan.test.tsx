/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import * as React from "react";
import { shallow } from "enzyme";
import { MessageSpan, MessageDiv } from "../../ui-framework/messages/MessageSpan";

describe("MessageSpan & MessageDiv", () => {

  describe("MessageSpan", () => {
    it("with message text", () => {
      shallow(<MessageSpan message="Test" />).should.matchSnapshot();
    });

    it("with message HTMLElement", () => {
      const newSpan = document.createElement("span");
      const newContent = document.createTextNode("Test");
      newSpan.appendChild(newContent);
      shallow(<MessageSpan message={newSpan} />).should.matchSnapshot();
    });
  });

  describe("MessageDiv", () => {
    it("with message text", () => {
      shallow(<MessageDiv message="Test" />).should.matchSnapshot();
    });

    it("with message HTMLElement", () => {
      const newSpan = document.createElement("span");
      const newContent = document.createTextNode("Test");
      newSpan.appendChild(newContent);
      shallow(<MessageDiv message={newSpan} />).should.matchSnapshot();
    });
  });

});
