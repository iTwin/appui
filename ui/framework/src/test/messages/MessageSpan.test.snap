// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageSpan & MessageDiv MessageDiv with React node 1`] = `
<div>
  <span>
    For more details, 
    <UnderlinedButton>
      click here
    </UnderlinedButton>
    .
  </span>
</div>
`;

exports[`MessageSpan & MessageDiv MessageDiv with message HTMLElement 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<span>Test</span>",
    }
  }
/>
`;

exports[`MessageSpan & MessageDiv MessageDiv with message text 1`] = `
<div>
  Test
</div>
`;

exports[`MessageSpan & MessageDiv MessageSpan with React node 1`] = `
<span>
  <span>
    For more details, 
    <UnderlinedButton>
      click here
    </UnderlinedButton>
    .
  </span>
</span>
`;

exports[`MessageSpan & MessageDiv MessageSpan with message HTMLElement 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<span>Test</span>",
    }
  }
/>
`;

exports[`MessageSpan & MessageDiv MessageSpan with message text 1`] = `
<span>
  Test
</span>
`;
